qemu/qemu.go

```diff
@@ -399,68 +453,104 @@ func (netdev NetDevice) Valid() bool {
 	}
 }
 
-// QemuParams returns the qemu parameters built out of this network device.
-func (netdev NetDevice) QemuParams(config *Config) []string {
-	var netdevParams []string
+// QemuDeviceParams returns the -device parameters for this network device
+func (netdev NetDevice) QemuDeviceParams(config *Config) []string {
 	var deviceParams []string
-	var qemuParams []string
 
-	if netdev.Driver == VirtioNetPCI {
-		deviceParams = append(deviceParams, "driver=")
-	}
-	deviceParams = append(deviceParams, fmt.Sprintf("%s", netdev.Driver))
-	if netdev.DisableModern {
-		deviceParams = append(deviceParams, ",disable-modern=true")
-	}
+	deviceParams = append(deviceParams, "driver=")
+	deviceParams = append(deviceParams, netdev.Type.QemuDeviceParam())
```

Comment:
Won't this generate an invalid cli in the case of `VHOSTUSER` (where `QemuDeviceParam()` returns `""`)?

Reply:
@jodh-intel great catch.. I  have yet to add/test that path. But this a generic issue. I could always check if we have params to append to as this code will get even more complex when more use cases come in.

Reply:
@markdryan @jodh-intel fixed.